lib/cosmos/io/json_drb.rb

```diff
@@ -90,7 +96,25 @@ def start_service(hostname = nil, port = nil, object = nil)
         @thread = Thread.new do
           begin
             while true
-              socket = @listen_socket.accept()
```

Comment:
accept blocks forever.  Need to use accept_nonblock and the self-pipe trick to be able to shutdown the thread cleanly.